File: crates/tracepilot-tauri-bindings/src/events.rs
//! Tauri IPC event name constants.
//!
//! Centralises event names emitted from Rust commands and listened to
//! on the TypeScript side.  Keep in sync with `packages/types/src/ipc-events.ts`.

/// Indexing lifecycle events (full index).
pub const INDEXING_STARTED: &str = "indexing-started";
pub const INDEXING_PROGRESS: &str = "indexing-progress";
pub const INDEXING_FINISHED: &str = "indexing-finished";

/// Copilot SDK IPC event (bridge event forwarding).
pub const SDK_BRIDGE_EVENT: &str = "sdk-bridge-event";
/// SDK connection state changed.
pub const SDK_CONNECTION_CHANGED: &str = "sdk-connection-changed";

/// Search-specific indexing events.
pub const SEARCH_INDEXING_STARTED: &str = "search-indexing-started";
pub const SEARCH_INDEXING_PROGRESS: &str = "search-indexing-progress";
pub const SEARCH_INDEXING_FINISHED: &str = "search-indexing-finished";

File: packages/client/src/events.ts
/**
 * Tauri IPC event name constants.
 *
 * Centralises event names emitted from Rust commands and listened to
 * on the TypeScript side.  Keep in sync with
 * `crates/tracepilot-tauri-bindings/src/events.rs`.
 *
 * Lives alongside `IPC_COMMANDS` (see `./commands.ts`) so that both
 * halves of the IPC wire-protocol surface have a single source of
 * truth in `@tracepilot/client`.
 */
export const IPC_EVENTS = {
  INDEXING_STARTED: "indexing-started",
  INDEXING_PROGRESS: "indexing-progress",
  INDEXING_FINISHED: "indexing-finished",
  SEARCH_INDEXING_STARTED: "search-indexing-started",
  SEARCH_INDEXING_PROGRESS: "search-indexing-progress",
  SEARCH_INDEXING_FINISHED: "search-indexing-finished",
  SDK_BRIDGE_EVENT: "sdk-bridge-event",
  SDK_SESSION_STATE_CHANGED: "sdk-session-state-changed",
  SDK_CONNECTION_CHANGED: "sdk-connection-changed",
} as const;

export type IpcEventName = (typeof IPC_EVENTS)[keyof typeof IPC_EVENTS];
